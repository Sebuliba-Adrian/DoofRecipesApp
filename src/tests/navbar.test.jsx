import React from "react";
import renderer from "react-test-renderer";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Navbar from "../components/navigation";

const props = {
  request: jest.fn(),
  viewCategories: jest.fn(),
  search: jest.fn()
};

global.localStorage = {
  setItem: () => {},
  getItem: () => {}
};
describe("Navbar tests", () => {
  configure({ adapter: new Adapter() });
  const navbar = mount(<Navbar {...props} />);
  const instance = navbar.instance();

  it("renders correctly", () => {
    const navBar = renderer.create(<Navbar {...props} />).toJSON();
    expect(navBar).toMatchSnapshot();
  });

  it("Logs out user", () => {
    const logoutLink = navbar.find("a").at(1);
    logoutLink.simulate("click");
    expect(props.request).toHaveBeenCalled();
  });

  it("Shows categories", () => {
    const logo = navbar.find("a").at(0);
    logo.simulate("click");
    expect(props.viewCategories).toHaveBeenCalled();
  });

  it("Triggers search", () => {
    const searchBox = navbar.find("[type='text']");
    searchBox.simulate("keyDown", { keyCode: 13 });
    expect(props.search).toHaveBeenCalled();
  });
});
