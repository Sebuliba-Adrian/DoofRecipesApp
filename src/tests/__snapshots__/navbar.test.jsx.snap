// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests renders correctly 1`] = `
<div
  className=""
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      <a
        className="navbar-brand logo text-white"
        href="#/"
        onClick={[Function]}
      >
        <h3>
          <span
            className="fa fa-lg fa-cutlery mr-3 ml-3"
          />
          |   Doof! recipes
        </h3>
      </a>
    </div>
    <div
      className="col-md-4 text-center"
    >
      <input
        className="form-control mr-sm-2 search-box"
        onKeyDown={[Function]}
        placeholder="Search for categories here..."
        type="text"
      />
    </div>
    <div
      className="col-md-4 text-right"
    >
      <div
        className="dropdown logout-dropdown"
      >
        <div
          className="fa fa-user-circle fa-lg text-white dropdown-toggle"
          data-toggle="dropdown"
          id="dropdownMenuLink"
        >
           
          <span
            className="logo"
          >
            <b>
               
               
            </b>
          </span>
        </div>
        <div
          aria-labelledby="dropdownMenuLink"
          className="dropdown-menu dropdown-cust"
        >
          <a
            className="dropdown-item"
            href="#/"
            id="logoutLink"
            onClick={[Function]}
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
